cmd/vela-git/main.go

```diff
@@ -73,6 +73,12 @@ func main() {
 			Usage:    "commit reference to clone from the repo",
 			Value:    "refs/heads/master",
 		},
+		&cli.StringFlag{
+			EnvVars:  []string{"PARAMETER_DEPTH", "GIT_DEPTH"},
+			FilePath: "/vela/parameters/git/depth,/vela/secrets/git/depth",
+			Name:     "build.depth",
+			Usage:    "enables fetching the repository with the specified depth",
+		},
```

Comment:
Should depth be a string flag? I wonder if it should be an int flag instead?

Thinking that typically `depth` in terms of the `git` CLI is represented by a number (1, 2, 10 etc)

Reply:
Originally I utilized `IntFlag` for the depth since I agree with you. However, we only utilize that parameter within the [exec.Command](https://golang.org/pkg/os/exec/#Command) and it only accepts a slice of strings. I don't personally see any additional value in converting the int to string when we utilize the parameter.

Reply:
Hmm... I agree the value proposition is low since it really won't make a difference in functionality.

So I think it's fair to keep it as a string flag 👍 

FWIW:

My preference would still be to treat it as an `IntFlag` and convert to a string when we need to

I guess for no other reason, it is "more consistent" with the upstream `git` CLI but the value again is weak